woodwork/statistics_utils/frequency_inference/_infer_frequency.py

```diff
@@ -0,0 +1,135 @@
+import dataclasses
+from ._generate_freq_candidates import _generate_freq_candidates
+from ._determine_most_likely_freq import _determine_most_likely_freq
+from ._generate_estimated_timeseries import _generate_estimated_timeseries
+from ._determine_missing_values import _determine_missing_values
+from ._determine_duplicate_values import _determine_duplicate_values
+from ._determine_nan_values import _determine_nan_values
+from ._determine_extra_values import _determine_extra_values
+from ._clean_timeseries import _clean_timeseries
+from ._types import InferDebug, DataCheckMessageCode
+
+import pandas as pd
+
+
+def inference_response(inferred_freq, debug_obj, debug):
+    if debug:
+        return (
+            inferred_freq,
+            dataclasses.asdict(debug_obj)
+        )
+    else:
+        return inferred_freq
+
+def infer_frequency(observed_ts: pd.Series, debug=False):
+    """Infer the frequency of a given Pandas Datetime Series.
+
+    Args:
+        series (pd.Series): data to use for histogram
+
+    Returns:
+        inferred_freq: a string
+        debug: a dictionary containing debug information if frequency cannot be inferred
+    """
+
+    pandas_inferred_freq = pd.infer_freq(observed_ts)
+
+    if pandas_inferred_freq:
+        return inference_response(
+            inferred_freq=pandas_inferred_freq,
+            debug_obj=InferDebug(),
+            debug=debug
+        )
+
+    # clean observed timeseries from duplicates and NaTs
+    observed_ts_clean = _clean_timeseries(observed_ts)
+
+    # Determine if series is not empty
+    if len(observed_ts_clean) == 0:
+        return inference_response(
+            inferred_freq=None,
+            debug_obj=InferDebug(
+                message = DataCheckMessageCode.DATETIME_SERIES_IS_EMPTY,
+            ),
+        )
+
+    nan_values = _determine_nan_values(observed_ts)
+    duplicate_values = _determine_duplicate_values(observed_ts)
+
+    actual_range_start = observed_ts_clean.min().isoformat()
+    actual_range_end = observed_ts_clean.max().isoformat()
```

Comment:
Since we check that the series is monotonic increasing, can we just take the value at the 0th and nth index?

Reply:
Or does that stop them from being "Actual" min and max values?

Reply:
no, I think you are right, good catch